pkg/registry/batch/cronjob/storage/storage.go

```diff
@@ -84,3 +101,53 @@ func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *me
 func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
 	return r.store.Update(ctx, name, objInfo)
 }
+
+// InstantiateREST implements the REST endpoint for manually triggering a CronJob.
+type InstantiateREST struct {
+	cronJobStore *genericregistry.Store
+	jobStore     *genericregistry.Store
+}
+
+var _ = rest.NamedCreater(&InstantiateREST{})
+
+func (r *InstantiateREST) New() runtime.Object {
+	return &batch.CronJobManualInstantiation{}
+}
+
+func (r *InstantiateREST) Create(ctx genericapirequest.Context, name string, obj runtime.Object, includeUninitialized bool) (out runtime.Object, err error) {
+	sourceCronJobObj, err := r.cronJobStore.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	sourceCronJob, ok := sourceCronJobObj.(*batch.CronJob)
+	if !ok {
+		return nil, fmt.Errorf("got object of type %T, expected *batch.CronJob", sourceCronJobObj)
+	}
+
+	job := &batch.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        fmt.Sprintf("%s-manual-%d", name, time.Now().Unix()),
+			Labels:      sourceCronJob.Spec.JobTemplate.Labels,
+			Annotations: sourceCronJob.Spec.JobTemplate.Annotations,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(sourceCronJob, batch.SchemeGroupVersion.WithKind("CronJob")),
+			},
+		},
+		Spec: sourceCronJob.Spec.JobTemplate.Spec,
+	}
+
+	createdJobObj, err := r.jobStore.Create(ctx, job, includeUninitialized)
+	if err != nil {
+		return nil, err
+	}
+	createdJob, ok := createdJobObj.(*batch.Job)
+	if !ok {
+		return nil, fmt.Errorf("got object of type %T, expected *batch.Job", createdJobObj)
+	}
+
+	// TODO update the CronJob so that it knows about the instantiated job
+	// not currently done because clobbering the CronJob in the store would create a race condition that
+	// could confuse the CronJob controller, which expects to have exclusive control over a CronJob's status
```

Comment:
I think this TODO is addressed in the 3rd commit so it can be removed from here.